test/metabase/actions_test.clj

```diff
@@ -477,3 +485,96 @@
                     [2 "American"]
                     [3 "Artisan"]]
                    (first-three-categories)))))))))
+
+(defn basic-auth-ssh-server ^java.io.Closeable [username password]
+  (try
+    (let [password-auth    (reify org.apache.sshd.server.auth.password.PasswordAuthenticator
+                             (authenticate [_ auth-username auth-password _session]
+                               (and
+                                (= auth-username username)
+                                (= auth-password password))))
+          keypair-provider (org.apache.sshd.server.keyprovider.SimpleGeneratorHostKeyProvider.)
+          sshd             (doto (SshServer/setUpDefaultServer)
+                             (.setPort 0)
+                             (.setKeyPairProvider keypair-provider)
+                             (.setPasswordAuthenticator password-auth)
+                             (.setForwardingFilter AcceptAllForwardingFilter/INSTANCE)
+                             .start)]
+      sshd)
+    (catch Throwable e
+      (throw (ex-info (format "Error starting SSH mock server with password")
+                      {:username username :password password}
+                      e)))))
+
+(deftest actions-on-ssh-tunneled-db
+  ;; testing actions against dbs with ssh tunnels. Use an in-memory ssh server that just forwards to the correct port
+  ;; through localhost. Since it is local, it's possible for the application to ignore the ssh tunnel and just talk to
+  ;; the db on the correct port. So we do the tests twice, once expecting the correct answer, and then again with the
+  ;; wrong password for the ssh tunnel and we expect failures. This ensures that the correct result is actually going
+  ;; through the ssh tunnel
+  (mt/test-drivers (disj (mt/normal-drivers-with-feature :actions) :h2)
```

Comment:
Do we support ssh tunnelling with H2? In theory someone could run an H2 database in server mode and connect to it with an SSH tunnel. See `http://h2database.com/html/features.html#auto_mixed_mode`. 

I see code supporting it [here](https://github.com/metabase/metabase/blob/26656c26ad10ed3581eb40cd682ab37cd7e612c6/src/metabase/driver/h2.clj#L496), but the "Use an SSH-tunnel" option is missing from the admin connection settings for H2 databases.

Reply:
I don't think we do. It throws an error in the tests because it's checking the host of the db details. As far as I know, h2 doesn't (ever?) have a host and it fails. To extend ssh to h2 would be a lot of work that doesn't need to happen now for this, and I suspect there isn't much demand for it anyways.